pkg/config/config.go

```diff
@@ -667,15 +672,18 @@ func LoadWithDefaults(getCfgPaths PathsGetter) (*Config, LoadWithDefaultsDetails
 	}, nil
 }
 
-// FromEnvOrFlag resolves and returns paths for config files.
+// FromProvider resolves and returns paths for config files.
 // It reads them the 'BOTKUBE_CONFIG_PATHS' env variable. If not found, then it uses '--config' flag.
-func FromEnvOrFlag() []string {
-	envCfgs := os.Getenv("BOTKUBE_CONFIG_PATHS")
-	if envCfgs != "" {
-		return strings.Split(envCfgs, ",")
+func FromProvider(gql *config.GqlClient) ([]string, error) {
+	var provider config.Provider
+	if os.Getenv("CONFIG_SOURCE_IDENTIFIER") != "" {
+		provider = config.NewGqlProvider(gql)
+	} else if os.Getenv("BOTKUBE_CONFIG_PATHS") != "" {
+		provider = config.NewEnvProvider()
+	} else {
+		provider = config.NewStaticProvider(configPathsFlag)
 	}
```

Comment:
I suggest refactoring the code a bit to reduce indentations:

```go
if ... {
    return ...
}

if ... {
    return ...
}
```

BTW you can already return `(Config, error)` struct instead of file paths. That is related to my previous comment about reading raw bytes directly by koanf 👍

Reply:
I would but I am using `provider` variable at the end of the function.